상품 목록 API 연동 및 필터 구현

HelloHailie opened the pull request:
## #️⃣ Issue Number

<!--- ex) #이슈번호, #이슈번호 -->

## 📝 요약(Summary)

'/products' 화면에서 확인할 수 있습니다. 

- 상품 목록 API 연동 및 MSW 핸들러 구현했습니다.
- 필터 옵션 수정 및 타입 정의 추가했습니다.

<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)
![스크린샷 2025-01-21 오후 10 58 21](https://github.com/user-attachments/assets/9f973a68-fcc8-4858-bbaa-3cd78344c306)



## 💬 공유사항 to 리뷰어

작업 내용은 아래와 같습니다. 
- ProductApis에 상품 목록 API 엔드포인트 추가
- MSW에 상품 목록 API 모킹 핸들러 구현
- 하드코딩된 상품 데이터를 API 호출로 대체
- 색상 필터 옵션을 API 응답과 일치하도록 수정
- Product 인터페이스 정의 추가


아래 사항을 확인 부탁드려요.

- 필터 기능 정상 동작 확인
- MSW 핸들러 동작 확인

<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

HelloHailie committed:
feat: 상품 목록 API 연동 및 필터 옵션 수정

- 상품 목록 API 엔드포인트 추가 및 MSW 핸들러 구현
- 상품 목록 페이지를 서버 컴포넌트로 변경하여 데이터 페칭 구현
- 필터 옵션의 색상값을 실제 API 응답값과 일치하도록 수정
- 상품 타입 정의 추가
- 목업 데이터 제거

HelloHailie committed:
fix: 최신 데이터로 수정 후 영향 범위 수정

HelloHailie commented:
@kigpand 말씀주신대로 최신 데이터 추가하여 수정하였습니다. 시간 되실때 확인 부탁드려요. 감사합니다. 

https://app.swaggerhub.com/apis/SMJ9030_1/Product-MVP/1.0.0#/default/get_api_v1_products

kigpand commented on src/app/products/page.tsx:
혹시 만들어져있는 getProducts 사용안하시고 따로 만들어서 사용하신 이유가 있을까요? 혹시 함수를 못찾아서 그러신거면 src/api/product.ts에 있습니다!

HelloHailie replied:
노티 주셔서 감사합니다. 만들어주신 함수 사용하도록 수정하고, 그 외 더이상 사용하지 않는 코드들 제거 처리하였습니다. 
시간 되시면 제가 사용한 방법이 맞는지 확인 부탁드릴게요! 더 좋은 방법이 있다면 알려주세요. 😊

kigpand replied:
@HelloHailie 확인했습니다! 고생하셨어요!! 😁

HelloHailie committed:
fix: 만들어져있는 getProducts 사용하도록 수정

- 사용하지 않는 mockData 제거 및 실제 API 연동
- 상품 목록 조회 시 카테고리, 이름, 평점, 정렬 기준 적용
- 타입 정의 통합 (Product -> IProduct)

HelloHailie committed:
feat: 상품 목록 페이지 UI/UX 개선

HelloHailie committed:
chore: Merge branch 'develop' into feature/product-swr
